nrf-hal-common/src/twis.rs

```diff
@@ -278,6 +395,66 @@ where
         Ok(())
     }
 
+    /// Receives data into the given `buffer`. Buffer must be located in RAM.
+    /// Returns a value that represents the in-progress DMA transfer.
+    #[allow(unused_mut)]
+    pub fn rx<W, B>(mut self, mut buffer: B) -> Result<Transfer<T, B>, Error>
```

Comment:
This seems to rely on `Drop` being called, right? If a transfer is started and then leaked, the caller could access the original `buffer` again if I'm not mistaken (if it passes a borrowed buffer type like `&mut [T]`).

Reply:
Goos point, I'll add a + 'static lifetime bound on the buffer to deal with this!